File: SoftInvModel/src/main/java/pe/edu/pucp/softinv/model/Pedido/DetallePedidoDTO.java
package pe.edu.pucp.softinv.model.Pedido;

import pe.edu.pucp.softinv.model.Producto.ProductoDTO;

public class DetallePedidoDTO {
    private ProductoDTO producto;
    private PedidoDTO pedido;
    private Integer cantidad;
    private Double subtotal;

    public PedidoDTO getPedido() {
        return pedido;
    }

    public void setPedido(PedidoDTO pedido) {
        this.pedido = pedido;
    }

    public ProductoDTO getProducto() {
        return producto;
    }

    public void setProducto(ProductoDTO producto) {
        this.producto = producto;
    }

    public Integer getCantidad() {
        return cantidad;
    }

    public void setCantidad(Integer cantidad) {
        this.cantidad = cantidad;
    }

    public Double getSubtotal() {
        return subtotal;
    }

    public void setSubtotal(Double subtotal) {
        this.subtotal = subtotal;
    }

    public DetallePedidoDTO() {
        producto = null;
        cantidad = null;
        subtotal = null;
        pedido = null;
    }

    public DetallePedidoDTO(ProductoDTO producto, PedidoDTO pedido, Integer cantidad, Double subtotal) {
        this.producto = producto;
        this.cantidad = cantidad;
        this.subtotal = subtotal;
        this.pedido = pedido;
    }
}

File: C#/SoftInvModel/Pedido/DetallePedidoDTO.cs
using System;

namespace softinv.model
{
    public class DetallePedidoDTO
    {
        private ProductoDTO producto;
        private PedidoDTO pedido;
        private int? cantidad;
        private double? subtotal;


        public DetallePedidoDTO()
        {
            Producto = null;
            Cantidad = null;
            Subtotal = null;
            Pedido = null;
        }

        public DetallePedidoDTO(ProductoDTO producto, PedidoDTO pedido, int cantidad, double subtotal)
        {
            this.Producto = producto;
            this.Cantidad = cantidad;
            this.Subtotal = subtotal;
            this.Pedido = pedido;
            //COMPLETAR POR FAVOR
            //this.
        }

        public ProductoDTO Producto { get => producto; set => producto = value; }
        public PedidoDTO Pedido { get => pedido; set => pedido = value; }
        public int? Cantidad { get => cantidad; set => cantidad = value; }
        public double? Subtotal { get => subtotal; set => subtotal = value; }
    }
}
